### PR TITLE
Changes needed to test Ldap authentication on containerised RGW

### PR DESCRIPTION
**Pull request by TejasC88**
Test to Integrate RGW with LDAP authentication, its partly in this script and partly in CephCI.

Pass log : http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/ldap_auth.txt

### CODE REVIEW

File: rgw/v2/tests/aws/test_ldap_auth.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    # check for any crashes during the execution
+    crash_info = s3_reusable.check_for_crash()
+    if crash_info:
+        raise TestExecError("ceph daemon crash found!")
+
+
+if __name__ == "__main__":
+
+    test_info = AddTestInfo("test bucket creation through awscli")
```
Comment (mkasturi18) [inline rgw/v2/tests/aws/test_ldap_auth.py:90]: @TejasC88 , can you fix the testinfo

File: rgw/v2/tests/aws/test_ldap_auth.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        bucket_name = "ldap" + str(bc)
+        cmd = f"AWS_ACCESS_KEY_ID={token_str} AWS_SECRET_ACCESS_KEY=' ' /usr/local/bin/aws s3 mb s3://{bucket_name} --endpoint http://{rgw_ip}:{rgw_port}"
+        out = utils.exec_shell_cmd(cmd)
+        log.info("Bucket created: " + bucket_name)
+
+    # check for any crashes during the execution
+    crash_info = s3_reusable.check_for_crash()
+    if crash_info:
+        raise TestExecError("ceph daemon crash found!")
+
```
Comment (mkasturi18) [inline rgw/v2/tests/aws/test_ldap_auth.py:86]: @TejasC88 , could you also include object creation and listing in this PR

**Review by mkasturi18 — CHANGES_REQUESTED** [review]

File: rgw/v2/tests/aws/test_ldap_auth.py (hunk 1 above)
Comment (TejasC88) [reply rgw/v2/tests/aws/test_ldap_auth.py]: Done.

File: rgw/v2/tests/aws/test_ldap_auth.py (hunk 2 above)
Comment (TejasC88) [reply rgw/v2/tests/aws/test_ldap_auth.py:94]: @mkasturi18  Object creation and bucket listing added in the script 
Pass log : http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/ldap_auth.txt

Can you pls review ?

**Review by mkasturi18 — APPROVED** [review]
Changes looks good

**Pull request merged**